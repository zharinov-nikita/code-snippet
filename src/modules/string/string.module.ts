export class ModuleString {
  public toCamelCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const firstWord = array[0]
    const otherWord = array.map((item, index) => {
      if (index > 0) {
        return `${item[0].toUpperCase()}${item.substring(1)}`
      }
    })

    return `${firstWord}${otherWord.join('')}`
  }

  public toUpperSnakeCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const string = array.map((item) => item.toUpperCase())
    return string.join('_')
  }

  public toLowerSnakeCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const string = array.map((item) => item.toLowerCase())
    return string.join('_')
  }

  public toUpperKebabCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    return stringInKebabCase.toUpperCase()
  }

  public toLowerKebabCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    return stringInKebabCase.toLowerCase()
  }

  public toPascalCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const firstWord = `${array[0][0].toUpperCase()}${array[0].substring(1)}`
    const otherWord = array.map((item, index) => {
      if (index > 0) {
        return `${item[0].toUpperCase()}${item.substring(1)}`
      }
    })

    return `${firstWord}${otherWord.join('')}`
  }

  private isFirstLetterUpperCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.charAt(0).toUpperCase() === stringInAnyCase.charAt(0)
  }

  private isFirstLetterLowerCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.charAt(0).toLowerCase() === stringInAnyCase.charAt(0)
  }

  private isStringContainsDash(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.search(/-/g) > 0
  }

  private isStringContainsUnderlining(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.search(/_/g) > 0
  }

  private isStringUpperCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.toUpperCase() === stringInAnyCase
  }

  private isStringLowerCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.toLowerCase() === stringInAnyCase
  }

  public isCamelCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isFirstLetterUpperCase(stringInAnyCase)) return false
    if (this.isStringContainsDash(stringInAnyCase)) return false
    if (this.isStringContainsUnderlining(stringInAnyCase)) return false
    if (this.isStringUpperCase(stringInAnyCase)) return false
    return true
  }

  public isPascalCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isFirstLetterLowerCase(stringInAnyCase)) return false
    if (this.isStringContainsDash(stringInAnyCase)) return false
    if (this.isStringContainsUnderlining(stringInAnyCase)) return false
    if (this.isStringUpperCase(stringInAnyCase)) return false
    return true
  }

  public isUpperSnakeCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringUpperCase(stringInAnyCase) && this.isStringContainsUnderlining(stringInAnyCase)) {
      return true
    }
    return false
  }

  public isLowerSnakeCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringLowerCase(stringInAnyCase) && this.isStringContainsUnderlining(stringInAnyCase)) {
      return true
    }
    return false
  }

  public isUpperKebabCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringUpperCase(stringInAnyCase) && this.isStringContainsDash(stringInAnyCase)) {
      return true
    }
    return false
  }

  public isLowerKebabCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringLowerCase(stringInAnyCase) && this.isStringContainsDash(stringInAnyCase)) {
      return true
    }
    return false
  }
}
