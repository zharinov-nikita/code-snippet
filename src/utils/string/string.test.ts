import { describe, expect, test } from '@jest/globals'
import { UtilString } from './string.util'

describe('util/string', () => {
  const util = new UtilString()
  const stringInKebabCase = 'hello-world'

  test('toCase', () => {
    expect(util.toCamelCase(stringInKebabCase)).toBe('helloWorld')
    expect(util.toLowerKebabCase(stringInKebabCase)).toBe('hello-world')
    expect(util.toLowerSnakeCase(stringInKebabCase)).toBe('hello_world')
    expect(util.toPascalCase(stringInKebabCase)).toBe('HelloWorld')
    expect(util.toUpperKebabCase(stringInKebabCase)).toBe('HELLO-WORLD')
    expect(util.toUpperSnakeCase(stringInKebabCase)).toBe('HELLO_WORLD')
  })

  test('snapshot', () => {
    expect(util.toCamelCase(stringInKebabCase)).toMatchSnapshot('helloWorld')
    expect(util.toLowerKebabCase(stringInKebabCase)).toMatchSnapshot('hello-world')
    expect(util.toLowerSnakeCase(stringInKebabCase)).toMatchSnapshot('hello_world')
    expect(util.toPascalCase(stringInKebabCase)).toMatchSnapshot('HelloWorld')
    expect(util.toUpperKebabCase(stringInKebabCase)).toMatchSnapshot('HELLO-WORLD')
    expect(util.toUpperSnakeCase(stringInKebabCase)).toMatchSnapshot('HELLO_WORLD')
  })
})
