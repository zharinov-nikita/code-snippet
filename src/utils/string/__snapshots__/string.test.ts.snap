// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/string snapshot: HELLO_WORLD 1`] = `"HELLO_WORLD"`;

exports[`util/string snapshot: HELLO-WORLD 1`] = `"HELLO-WORLD"`;

exports[`util/string snapshot: HelloWorld 1`] = `"HelloWorld"`;

exports[`util/string snapshot: hello_world 1`] = `"hello_world"`;

exports[`util/string snapshot: hello-world 1`] = `"hello-world"`;

exports[`util/string snapshot: helloWorld 1`] = `"helloWorld"`;
